pkg/controller/controller.go

```diff
@@ -1141,6 +1143,20 @@ func (p *csiProvisioner) Delete(ctx context.Context, volume *v1.PersistentVolume
 	storageClassName := util.GetPersistentVolumeClass(volume)
 	if len(storageClassName) != 0 {
 		if storageClass, err := p.scLister.Get(storageClassName); err == nil {
+			if volumeMigrated {
+				inTreePluginName, err := p.translator.GetInTreeNameFromCSIName(volume.Spec.CSI.Driver)
+				if err != nil {
+					return err
+				}
+				
+				if inTreePluginName == storageClass.Provisioner {
```

Comment:
Can we directly use `if storageClass.Provisioner == p.supportsMigrationFromInTreePluginName`